app/controllers/concerns/git_interface.rb

```diff
@@ -2,69 +2,101 @@ module GitInterface
   extend ActiveSupport::Concern
   require 'rugged'
   
-  def rugged_create (id,string,action)
-    repo = setup
-    #find/create branch and check it out
-    branch = repo.branches[id]
-    if branch.nil?
-      branch = repo.branches.create(id, "HEAD")
-    end
-    repo.checkout(branch)
-    #add blob
-    oid = repo.write(string,:blob)
-    index = repo.index
-    index.read_tree(repo.head.target.tree)
-    index.add(:path => "#{id}.nt", :oid => oid, :mode => 0100644)
-    #commit
-    options = {}
-    options[:tree] = index.write_tree(repo)
-    options[:author] = {:email => current_user.email, :name => current_user.name, :time => Time.now }
-    options[:committer] = {:email => current_user.email, :name => current_user.name, :time => Time.now }
-    options[:message] = action + ": " + id
-    options[:parents] = repo.empty? ? [] : [ repo.head.target ].compact
-    options[:update_ref] = 'HEAD'
-    Rugged::Commit.create(repo, options)
-    index.write
-    options = {}
-    options[:strategy] = :force
-    repo.checkout_head(options)
-    repo.checkout('master')
-  end
-
-  def rugged_merge (id)
-    repo = setup
-    repo.checkout('master')
-    #merge
-    into_branch = 'master'
-    from_branch = id
-    their_commit = repo.branches[into_branch].target_id
-    our_commit = repo.branches[from_branch].target_id
-
-    merge_index = repo.merge_commits(our_commit, their_commit)
-
-    if merge_index.conflicts?
-      # conflicts. deal with them
-    else
-      # no conflicts
-      commit_tree = merge_index.write_tree(repo)
+  def rugged_create (id, branch_id, string, action)
+    begin
+      repo = setup
+      #find/create branch and check it out
+      branch = repo.branches[branch_id]
+      if branch.nil?
+        branch = repo.branches.create(branch_id, "HEAD")
+      end
+      repo.checkout(branch)
+      #add blob
+      oid = repo.write(string,:blob)
+      index = repo.index
+      index.read_tree(repo.head.target.tree)
+      index.add(:path => "#{id}.nt", :oid => oid, :mode => 0100644)
+      #commit
       options = {}
-      options[:tree] = commit_tree
-      options[:author] = { :email => current_user.email, :name => current_user.name, :time => Time.now }
-      options[:committer] = { :email => current_user.email, :name => current_user.name, :time => Time.now }
-      options[:message] ||= "Merge #{from_branch} into #{into_branch}"
-      options[:parents] = [repo.head.target, our_commit]
+      options[:tree] = index.write_tree(repo)
+      options[:author] = {:email => current_user.email, :name => current_user.name, :time => Time.now }
+      options[:committer] = {:email => current_user.email, :name => current_user.name, :time => Time.now }
```

Comment:
Im curious as to what the difference between the author and the committer is here. Just for clarification on my end.